ceph_iscsi_config/client.py

```diff
@@ -679,10 +680,15 @@ def _decrypt(self):
         key_path = os.path.join(settings.config.ceph_config_dir,
                                 settings.config.priv_key)
         try:
-            key = RSA.importKey(open(key_path))
-
-            cipher = PKCS1_OAEP.new(key)
-            plain_pw = cipher.decrypt(b64decode(self.password_str))
+            with open(key_path, 'rb') as keyf:
+                key = serialization.load_pem_private_key(keyf.read(), None,
+                                                         default_backend())
+
+            plain_pw = key.decrypt(b64decode(self.password_str),
```

Comment:
Nit: I might be wrong, but I think the older library might have defaulted to the insecure SHA-1 hash. If the CHAP password was previously encrypted w/ the SHA-1 hash, would it fail to decode here? If so, for initial upgrade backwards compatibility, should it catch ```ValueError``` and try again w/ the downgrade?

Reply:
I think you might be right. I forgot about the upgrade path. I'll add a unit test that encrypts the password using the previous library (python-crypto) and decrypt with the new code. That will give us some confidence.